File: Medium/0238. Product of Array Except Self/0238. Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        output = [1] * len(nums)

        for i in range(1, len(nums)):
            output[i] = output[i-1] * nums[i-1]

        right = 1
        for i in range(len(nums)-2, -1, -1):
            right = right * nums[i+1]
            output[i] = output[i] * right

        return output

File: Medium/0238. Product of Array Except Self/0238. Product of Array Except Self.go
func productExceptSelf(nums []int) []int {

    length := len(nums)
    out := make([]int, length)

    left := 1
    for i := 0; i < length; i++ {
        out[i] = left
        left *= nums[i]
    }

    right := 1
    for i := length-1; i >= 0; i-- {
        out[i] *= right
        right *= nums[i]
    }

    return out
}


